tests/integration_tests.rs

```diff
@@ -865,3 +865,43 @@ async fn test_flush_all() {
     let result = client.get(key).await;
     assert!(matches!(result, Ok(None)));
 }
+
+#[ignore = "Relies on a running memcached server"]
+#[tokio::test]
+#[parallel]
+async fn test_gat_with_nonexistent_key() {
+    let key = "nonexistent-key";
+
+    let mut client = setup_client(&[key]).await;
+
+    let get_result = client.gat(0, key).await;
+
+    assert!(matches!(get_result, Ok(None)), "key should not be found");
+}
+
+#[ignore = "Relies on a running memcached server"]
+#[tokio::test]
+#[parallel]
+async fn test_gat_with_cached_key() {
```

Comment:
A big part of `gat` is the updated expiration time. A test should be added to make sure that is working correctly.

Reply:
Done